### PR TITLE
Highlight crate links like normal links

### LINKED ISSUE #75823: Highlight crate links as normal links with --enable-index-page
Rustdoc highlights most links in blue, for example module links:

image

But it does not highlight the links on the docs index page:

image

Generally links are highlighted in blue (or some other color) on websites and the rest of Rustdoc highlights links so it would be more consistent to highlight these links.

### PR DESCRIPTION
**Pull request by camelid**
Fixes #75823.

Cc @jyn514

### CODE REVIEW

File: src/librustdoc/html/render/mod.rs [hunk 1]
```diff
@@ -1066,7 +1066,7 @@ themePicker.onblur = handleThemeButtonsBlur;
                 krates
                     .iter()
                     .map(|s| {
-                        format!("<li><a href=\"{}index.html\">{}</li>", ensure_trailing_slash(s), s)
+                        format!("<li><a class=\"mod\" href=\"{}index.html\">{}</a></li>", ensure_trailing_slash(s), s)
```
Comment (camelid) [inline src/librustdoc/html/render/mod.rs:1069]: Using class `mod` should probably be temporary (I should probably be a separate class `crate`). However, the `ul` that this is in (starting on line 1065) already has class `mod`!

**Comment by camelid** [comment]
I'm waiting on a rustc build to see if this works - for some reason it's rebuilding LLVM... :(

**Comment by camelid** [comment]
Fixed formatting

File: src/librustdoc/html/render/mod.rs [hunk 2]
```diff
@@ -1066,7 +1066,11 @@ themePicker.onblur = handleThemeButtonsBlur;
                 krates
                     .iter()
                     .map(|s| {
-                        format!("<li><a href=\"{}index.html\">{}</li>", ensure_trailing_slash(s), s)
+                        format!(
+                            "<li><a class=\"mod\" href=\"{}index.html\">{}</a></li>",
+                            ensure_trailing_slash(s),
+                            s
+                        )
```
Comment (camelid) [inline src/librustdoc/html/render/mod.rs:1073]: Using class mod should probably be temporary (I should probably be a separate class crate). However, the ul that this is in (starting on line 1065) already has class mod!

Comment (camelid) [reply src/librustdoc/html/render/mod.rs:1073]: I added a missing closing `</a>` tag also.

**Comment by jyn514** [comment]
I think ollie might actually have been a better reviewer, I don't know this part of the code very well ... let's

r? @GuillaumeGomez

since he has so much free time ;)

**Review by GuillaumeGomez — APPROVED** [review]

**Comment by GuillaumeGomez** [comment]
Looks good to me. Is there anything else to be done here?

> since he has so much free time ;)

:stuck_out_tongue_closed_eyes:

**Comment by camelid** [comment]
I think I should add a new CSS class `crate` instead of using `mod`.

**Comment by camelid** [comment]
Actually, since we want the styles to be the same, it might be better to not add a separate class since it might get out of date if the module styles are updated.

**Comment by camelid** [comment]
I wasn't able to verify that the styling looks correct since I've been having some issues building the compiler (https://rust-lang.zulipchat.com/#narrow/stream/182449-t-compiler.2Fhelp/topic/ICE.20when.20compiling).

**Comment by camelid** [comment]
One final thing which is somewhat unrelated is that the title, description, and heading for the index page are all different:

https://github.com/rust-lang/rust/pull/75842/commits/4419e04f16713d85fca2769c88e2677c68fb75e9#diff-32f26f9399b14283b32de5800d99cb12R1047-R1057

It might be good to consolidate them, but that could be done in a separate PR.

File: src/librustdoc/html/render/mod.rs [hunk 3]
```diff
@@ -1066,7 +1066,11 @@ themePicker.onblur = handleThemeButtonsBlur;
                 krates
                     .iter()
                     .map(|s| {
-                        format!("<li><a href=\"{}index.html\">{}</li>", ensure_trailing_slash(s), s)
+                        format!(
+                            "<li><a class=\"mod\" href=\"{}index.html\">{}</a></li>",
```
Comment (pickfire) [inline src/librustdoc/html/render/mod.rs:1070]:
```suggestion
                            "<li><a class='mod' href='{}index.html'>{}</a></li>",
```

File: src/librustdoc/html/render/mod.rs (hunk 2 above)
Comment (pickfire) [reply src/librustdoc/html/render/mod.rs:1073]: In the other link `class="mod"` was also added.

File: src/librustdoc/html/render/mod.rs (hunk 3 above)
Comment (GuillaumeGomez) [reply src/librustdoc/html/render/mod.rs:1070]: We're using double quotes everywhere else, so for consistency, I think we should keep them.

Comment (pickfire) [reply src/librustdoc/html/render/mod.rs:1070]: I saw that we also use single quotes for quite a lot of place too.

Comment (GuillaumeGomez) [reply src/librustdoc/html/render/mod.rs:1070]: 179 single quotes and 230 double quotes (looking for `='` and "=\"`). I wonder how we ended up in this mess... We should clean that up.

Comment (pickfire) [reply src/librustdoc/html/render/mod.rs:1070]: Why not just make all single quotes? We don't need to escape that way..

Comment (pickfire) [reply src/librustdoc/html/render/mod.rs:1070]: We could change that to
```
<div onclick="func('clicked');">...</div>
```

Comment (GuillaumeGomez) [reply src/librustdoc/html/render/mod.rs:1070]: Or to:

```
<div onclick='func("clicked");'>...</div>
```

:stuck_out_tongue:

**Comment by camelid** [comment]
@GuillaumeGomez I think this PR is ready. Do you want to merge? @pickfire or someone else could do a follow-up PR to make the quote character consistent in this file if that's what's decided upon.

**Comment by jyn514** [comment]
> I think this PR is ready.
> I'm waiting on a rustc build to see if this works - for some reason it's rebuilding LLVM... :(

Can you post a screenshot of what this now looks like?

**Comment by GuillaumeGomez** [comment]
Yes, just a screenshot before/after and then I think we're good to go!

**Comment by camelid** [comment]
Still waiting on my build :)

The before is at https://doc.rust-lang.org/nightly/nightly-rustc:

image

**Comment by camelid** [comment]
Wow, my build is taking forever. I just stopped and started it in case that’s the issue. `./x.py doc --stage 1` is the command to use, right?

**Comment by jyn514** [comment]
> Wow, my build is taking forever. I just stopped and started it in case that’s the issue. `./x.py doc --stage 1` is the command to use, right?

Yes, that sounds right. Try running with `-j1` if you aren't already, you might be thrashing memory to disk and back. It takes me about an hour on my laptop with -j1.

**Comment by camelid** [comment]
I have several gigabytes of free memory though, so would that still help?

**Comment by jyn514** [comment]
Looks good to me!
image

@bors r+

**Comment by camelid** [comment]
Never could get my build to finish :(

**Comment by GuillaumeGomez** [comment]
@camelid In case you want to retry to build doc locally: `x.py doc library/std --stage 1 --jobs 2` (`--jobs 2` is just to prevent your computer to burn too much haha). Then it's generated in `build/[your arch]/doc/std/index.html`.

**Comment by camelid** [comment]
@GuillaumeGomez Thanks, I'll try that!

**Comment by camelid** [comment]
The build completed successfully, but that index page is not the crate index, so it is not affected by this PR. I think I have to document the compiler to get the right page.

**Comment by jyn514** [comment]
@camelid try `compiler-docs = true` in `config.toml`

**Comment by camelid** [comment]
@jyn514 I'm curious: is there a reason that is not the default?

**Comment by jyn514** [comment]
Not sure, sorry. You could ask on zulip.

My guess though is that most of the compiler docs are maintained out of tree at rustc-dev-guide.rust-lang.org so people don't look at the API docs as much? I use the online ones a lot though.

**Pull request merged**